crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -3121,6 +3121,51 @@ impl<TX: DbTx + 'static, N: NodeTypesForProvider> StatsReader for DatabaseProvid
     }
 }
 
+impl<TX: DbTxMut + DbTx + 'static, N: NodeTypesForProvider> TransactionHashNumbersWriter
+    for DatabaseProvider<TX, N>
+{
+    fn insert_transaction_hash_numbers_raw<I>(&self, hash_to_number_iter: I) -> ProviderResult<bool>
+    where
+        I: Iterator<Item = std::io::Result<(Vec<u8>, Vec<u8>)>>,
+    {
+        #[cfg(all(unix, feature = "rocksdb"))]
+        if self.cached_storage_settings().transaction_hash_numbers_in_rocksdb {
+            use reth_db_api::table::Decompress;
+
+            self.rocksdb_provider().write_batch(|batch| {
+                for entry in hash_to_number_iter {
+                    let (hash_bytes, number_bytes) = entry.map_err(ProviderError::other)?;
+
+                    let hash =
+                        TxHash::try_from(hash_bytes.as_slice()).map_err(ProviderError::other)?;
+                    let number: TxNumber = Decompress::decompress(&number_bytes)?;
+
+                    batch.put::<tables::TransactionHashNumbers>(hash, &number)?;
+                }
+                Ok(())
+            })?;
+
+            return Ok(false);
```

Comment:
we are always returning `Ok(false)` here, but in the trait fn doc we say `Returns `Ok(true)` if the table was empty before insertion, `Ok(false)` otherwise.` https://github.com/paradigmxyz/reth/pull/20414/files#diff-92d898a32e1356e566dd6266a1c516f269575be896a5e4d1b22e0106ca30f5cdR132 

this return value seems to be used only for logging, maybe we should update the doc?